### ISSUE TITLE
Automate releases to Docker Hub

### ISSUE THREAD

**Issue by ericcornelissen** [ericcornelissen/js-regex-security-scanner#1]

**Comment by ericcornelissen** [comment]
The automation implemented in #41 was unfortunately incorrect (release `v0.2.2` has been released manually). The [validations job](https://github.com/ericcornelissen/js-regex-security-scanner/blob/3c04f437b2b0472a59394e84b3acc64ec76151da/.github/workflows/release.yml#L60-L76) can't succeed because when the code is checked out - leading to [the `shescape (v1.5.9)` test failing](https://github.com/ericcornelissen/js-regex-security-scanner/actions/runs/3173821699/jobs/5169983698#step:6:15).

Like [the `test.yml::test` job](https://github.com/ericcornelissen/js-regex-security-scanner/blob/main/.github/workflows/test.yml#L40-L58), the validation job should use the `submodule` option of the checkout action:

```diff
       - name: Checkout Code
         uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          submodules: true
```

**Comment by ericcornelissen** [comment]
The automated release for v0.2.3 (following #73) [successfully released to Docker hub](https://github.com/ericcornelissen/js-regex-security-scanner/actions/runs/3315705931/jobs/5476643786), but [failed to automatically update the `v0` branch](https://github.com/ericcornelissen/js-regex-security-scanner/actions/runs/3315705931/jobs/5476643613).

As this issue is about automatically releasing to Docker Hub only, I'm closing this issue.
